module1-introduction-to-backend/r3.4-api-best-practices/README.md

```diff
@@ -0,0 +1,232 @@
+# API Best Practices
+
+Now that we have learnt to build our own APIs, in this lesson we will cover some best practices for API development.
+
+Although some of these suggestions may require going through more modules of the course before implementation, consider this as a guide that you can come back and refer to at any time as you progress through the course.
+
+## What is API design?
+API design is the collection of planning and architectural decisions you make when building an API. Your basic API design influences how well developers are able to consume it and even how they use it. Just like website design or product design, API design informs the user experience. Good API design principles meet initial expectations and continue to behave consistently and predictably.
+
+Frontend developers need good UI design before building the product for their end users. Similarly backend developers need good API design because the users in this case are the frontend developers who will integrate with your API to build the end-user facing product.
```

Comment:
Similarly,

Reply:
i'd add a sentence saying frontend developers who integrate with your API should not know how your API works under the hood, for example ask the students if they have ever used an API, they probably have no idea how its implemented (and that's how it should be). and violating that would be a pretty egregious mistake